src/config.rs

```diff
@@ -0,0 +1,178 @@
+            max_line_length: 90,
+        }
+    }
+}
+
+/// Represents the configuration for the Sprocket `check` and `lint` commands.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct CheckConfig {
+    /// Rule IDs to except from running.
+    pub except: Vec<String>,
+    /// Causes the command to fail if any warnings are reported.
+    pub deny_warnings: bool,
+    /// Causes the command to fail if any notes are reported.
+    pub deny_notes: bool,
+    /// Hide diagnostics with `note` severity.
+    pub hide_notes: bool,
+}
+
+/// Represents the configuration for the Sprocket `run` command.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct RunConfig {
+    /// Path to the engine configuration file.
+    pub config: PathBuf,
+}
+
+impl Config {
+    /// Create a new config instance by reading potential configurations.
+    pub fn new(path: Option<String>) -> Self {
+        // Check for a config file in the current directory
+        // Start a new Figment instance with default values
+        let mut figment = Figment::new().admerge(Serialized::from(Config::default(), "default"));
+
+        // Check XDG_CONFIG_HOME for a config file
+        // On MacOS, check HOME for a config file
+        #[cfg(target_os = "macos")]
+        {
+            if let Some(home) = dirs::home_dir() {
+                tracing::info!(
+                    "reading configuration from: {home:?}/.config/sprocket/sprocket.toml"
+                );
+                figment = figment.admerge(Toml::file(
+                    home.join(".config").join("sprocket").join("sprocket.toml"),
+                ));
+            }
+        }
+        #[cfg(not(target_os = "macos"))]
+        {
+            if let Some(xdg_config_home) = dirs::config_dir() {
+                tracing::info!(
+                    "reading configuration from XDG_CONFIG_HOME: \
+                     {xdg_config_home:?}/sprocket/sprocket.toml"
+                );
+                figment = figment.admerge(Toml::file(
+                    xdg_config_home.join("sprocket").join("sprocket.toml"),
+                ));
+            }
+        }
+
+        // Check PWD for a config file
+        if Path::exists(Path::new("sprocket.toml")) {
+            tracing::info!("reading configuration from PWD/sprocket.toml");
+            figment = figment.admerge(Toml::file("sprocket.toml"));
+        }
+
+        // If provided, check config file from environment
+        if let Ok(config_file) = env::var("SPROCKET_CONFIG") {
+            tracing::info!("reading configuration from SPROCKET_CONFIG: {config_file:?}");
+            figment = figment.admerge(Toml::file(config_file));
+        }
+
+        // If provided, check command line config file
+        if let Some(ref cli) = path {
+            tracing::info!("reading configuration from --config: {cli:?}");
```

Comment:
```suggestion
            tracing::info!("reading configuration from provided configuration file: {cli:?}");
```

I think this is more future proof rather than calling out specific command line arguments.